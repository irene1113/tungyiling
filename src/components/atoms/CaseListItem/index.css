.case-list-item {
	color: var(--black);

	& > *:not(:last-child) {
		margin-bottom: 10px;
	}

	& > .title {
		font-family: 'Asap', sans-serif;
		font-size: 22px;
		font-weight: 700;
	}

	& > .duration {
		font-size: 18px;
		font-family: 'Puritan', sans-serif;
	}

	& > .content {
		font-size: 14px;
		font-weight: 300;
	}

	& > .tags {
		display: flex;
		flex: auto;

		& > div {
			font-size: 12px;

			&:not(:first-child) {
				margin-left: 6px;
			}
		}
	}

	& > .link {
		display: inline-flex;
		height: 20px;
		width: 30px;
		font-size: 12px;
		font-weight: bold;
		color: var(--primary);
	}
}
