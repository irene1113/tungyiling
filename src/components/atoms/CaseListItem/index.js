import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const CaseListItem = ({ className, title = '', duration = '', content = '' }) => (
	<div className={classnames(styles.caseListItem, className)}>
		<div className={styles.title}>{title}</div>
		<div className={styles.duration}>{duration}</div>
		<div className={styles.content}>{content}</div>
	</div>
);

export default CaseListItem;
