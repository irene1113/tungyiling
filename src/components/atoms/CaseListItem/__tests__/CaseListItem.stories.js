import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import CaseListItem from 'components/atoms/CaseListItem';

const stories = storiesOf('atoms/CaseListItem', module);

stories.add('__interactive', () => (
	<CaseListItem
		title={text('title', 'ISC Reporting')}
		duration={text('duration', '2019 - 2020')}
		content={text(
			'content',
			'description description description description description description.',
		)}
	/>
));
