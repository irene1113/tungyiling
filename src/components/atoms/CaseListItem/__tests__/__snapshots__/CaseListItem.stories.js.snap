// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="caseListItem"
  >
    <div
      class="title"
    >
      ISC Reporting
    </div>
    <div
      class="duration"
    >
      2019 - 2020
    </div>
    <div
      class="content"
    >
      description description description description description description.
    </div>
  </div>
</DocumentFragment>
`;
